### ISSUE TITLE
[amp-pwa-reader] Structured Data is not shown on Structured Data Test Tool

### ISSUE THREAD

**Issue by andreban** [ampproject/amp-publisher-sample#82]
Structured Data Test Tool is unable to extract data from articles. 

<img width="1334" alt="screen shot 2017-07-24 at 11 36 04 am" src="https://user-images.githubusercontent.com/1733592/28519637-4b4a95c6-7064-11e7-8e3f-c047257b3f40.png">

Results from SDTT can be seen here:
https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Famp.cards%2Ftheguardian%2Fus%2Fus-news%2F2017%2Fjul%2F24%2Fjared-kushner-new-york-russia-money-laundering

**Comment by pbakaus** [comment]
This most likely cannot be supported in this sample, as it is a purely client-side demo – unless the SDTT supports JS?

**Comment by sebastianbenz** [comment]
Hm. Can we somehow fix this? Otherwise the sample is useless (or even worse misleading) for most news publishers and ecommerce devs.

**Comment by pbakaus** [comment]
We could fix this by deploying it on an actual server, with an actual backend, and use hybrid rendering. That should totally be the way to go for actual production apps. I don't have the bandwidth to look at this right now, but would be more than happy if somebody else could pick it up.

**Comment by sebastianbenz** [comment]
Why are we not using the amp-install-serviceworker based approach? First view is AMP which then installs the shell (at least for the article pages)?

**Comment by pbakaus** [comment]
Yes, that would be the ideal setup. Again, requires a server-side component, since this uses proxied Guardian pages. Also don't think the Guardian would be super comfortable with us just directly proxying all their AMP pages (maybe).

**Comment by andreban** [comment]
Could we add a section on the docs to make clear that this is not covered by the demo, and point to possible solutions.

**Comment by ithinkihaveacat** [comment]
@andreban The issue is mentioned in the [Caveats section of the README](https://github.com/ampproject/amp-publisher-sample/blob/master/amp-pwa-reader/README.md#caveats), do you think it should be more prominent?

In at least one place the site is described as "production-ready" (albeit simultaneous with "sample")--that might not be quite the right word:

<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">Meet the⚡️ ShadowReader, a production-ready PWA+AMP sample that uses real data from <a href="https://twitter.com/guardian">@guardian</a>. Learn all about it 👇  <a href="https://t.co/jpPEPXLWLZ">https://t.co/jpPEPXLWLZ</a></p>&mdash; AMP Project (@AMPhtml) <a href="https://twitter.com/AMPhtml/status/888181282168946688">July 20, 2017</a></blockquote>

**Comment by andreban** [comment]
There are 2 things that we can change on the current docs. 

> Search engines that don't understand JS won't be able to crawl the site (use hybrid rendering to address).

Even if the search engine supports Javascript, the Schema Data won't be available. We could append this information to the caveat.

> amp-pwa-reader: Vanilla JS, detailed and SEO-friendly single-page progressive web app that displays AMP documents via Shadow DOM

I think the mentioned caveats make it hard to say that the project is entirely SEO-friendly. We may want to remove the claim from this part of the docs, and explain better what works and what doesn't work on the caveats.

**Comment by pbakaus** [comment]
@andreban yes, both suggestions are fair. I'll take a look this afternoon and will amend the README.

**Comment by pbakaus** [comment]
This requires a hybrid client/server rendering approach, which I've signed amedina@ up to take a look at.
